src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Input/Commands/AbstractGenerateNuGetPackageCommand.cs

```diff
@@ -2,18 +2,35 @@
 
 using System;
 using System.Threading.Tasks;
+using Microsoft.VisualStudio.ProjectSystem.Build;
 using Microsoft.VisualStudio.ProjectSystem.Input;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
 
 namespace Microsoft.VisualStudio.ProjectSystem.VS.Input.Commands
 {
-    internal abstract class AbstractGenerateNuGetPackageCommand : AbstractSingleNodeProjectCommand
+    internal abstract class AbstractGenerateNuGetPackageCommand : AbstractSingleNodeProjectCommand, IVsUpdateSolutionEvents
     {
         private readonly IProjectThreadingService _threadingService;
+        private readonly IVsSolutionBuildManager2 _buildManager;
+        private readonly GeneratePackageOnBuildPropertyProvider _generatePackageOnBuildPropertyProvider;
 
-        protected AbstractGenerateNuGetPackageCommand(UnconfiguredProject unconfiguredProject, IProjectThreadingService threadingService)
+        protected AbstractGenerateNuGetPackageCommand(
+            UnconfiguredProject unconfiguredProject,
+            IProjectThreadingService threadingService,
+            SVsServiceProvider serviceProvider,
+            GeneratePackageOnBuildPropertyProvider generatePackageOnBuildPropertyProvider)
         {
+            Requires.NotNull(serviceProvider, nameof(serviceProvider));
```

Comment:
Just because we're using MEF right now doesn't mean that we won't use something else in the future. It's always a good idea to have contracts explicitly expressed in code, and this doesn't cost much in terms of computing power.